src/components/pages/group/common/custom-field/custom-field-edit-from.tsx

```diff
@@ -0,0 +1,179 @@
+        iconURL: Yup.string().nullable(),
+        dataType: Yup.string().required(),
+        fieldType: Yup.string().required(),
+        defaultValue: Yup.string().nullable(),
+        cfValues: Yup.lazy((value) => {
+          if (value) {
+            return Yup.array().of(
+              Yup.object().shape({
+                value: Yup.string().nullable(),
+                iconURL: Yup.string().nullable(),
+                notes: Yup.string().nullable()
+              })
+            );
+          }
+          return Yup.mixed().notRequired();
+        })
+      })
+    ),
+    defaultValues :{
+      isMandatory:editCustomFieldData[0].isMandatory,
+      allowedParticipation:editCustomFieldData[0].allowedParticipation
+    }
+  });
+
+  const handleFormSubmit = async (value) => {
+
+    const editPayload ={
+      //Not changeable
+      defaultValue:defaultValues["defaultValue"],
+      displayOrder:defaultValues["displayOrder"],
+      userGroupId:userGroupId,
+      "customFields":{
+        id: defaultValues["id"],
+        authorId:defaultValues["authorId"],
+        dataType:defaultValues["dataType"],
+        fieldType:defaultValues["fieldType"],
+
+        //changeable
+        name:value["name"],
+        units:value["units"],
+        iconURL:value["iconURL"],
+        notes:value["notes"],
+      },
+      cfValues: value["values"],
+      isMandatory:value["isMandatory"],
+      allowedParticipation:value["allowedParticipation"],
+    };
+
+    const { success, data } = await axcustomFieldEditDetails(userGroupId,
+      customFieldId , editPayload)
+
+    if (success) {
+      notification(t("group:custom_field.update.success"), NotificationType.Success);
+      setCustomFields(data);
+      setIsEdit(false);
+    } else {
+      notification(t("group:custom_field.update.failure"));
+    }
+  };
+
+    const defaultValues = editCustomFieldData.reduce((acc, item) => {
+        const { customFields: cfs, cfValues, allowedParticipation, isMandatory, ...others } = item;
+        acc = {
+          ...cfs,
+          allowedParticipation:allowedParticipation,
+          isMandatory:isMandatory,
+          values: cfValues ? [...cfValues.map((i) => ({ value: i.values, ...i }))] : [],
+          ...others
+        };
+      return acc;
+    }, DEFAULT_CUSTOMFIELD_VALUE);
+
+    console.warn(defaultValues)
+
+    useEffect(() => {
+      hForm.reset(defaultValues);
+      setShowOption(categoricalType.includes(defaultValues["fieldType"]));
+    }, []);
+
+  return (
+    <FormProvider {...hForm}>
+      <form onSubmit={hForm.handleSubmit(handleFormSubmit)} className="fade">
+        <Button
+          mb={4}
+          type="button"
+          onClick={() => setIsEdit(false)}
+          leftIcon={<ArrowBackIcon />}
+        >
+          {t("group:custom_field.back")}
+        </Button>
+        <SimpleGrid columns={{ base: 1, md: 4 }} spacing={{ md: 4 }}>
+          <Box gridColumn="1/4">
```

Comment:
please use <GridItem/> component instead of `gridColumn` props https://chakra-ui.com/docs/components/grid#spanning-columns